cli/internal/publish/images.go

```diff
@@ -0,0 +1,458 @@
+package publish
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	cloudquery_api "github.com/cloudquery/cloudquery-api-go"
+	"github.com/cloudquery/cloudquery/cli/internal/hub"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/text"
+	"github.com/yuin/goldmark/util"
+	"golang.org/x/exp/maps"
+	"golang.org/x/sync/errgroup"
+)
+
+type imageReference struct {
+	ref     string // image filename inc. all paths (to replace with URL)
+	absFile string // absolute path to image file, to upload
+	url     string // result of upload
+
+	startPos, endPos int // start,end of complete markdown tag. if html: start,end of actual ref
+}
+
+var htmlImageRe = regexp.MustCompile(`<img\s+(?:[^>"']|"[^"]*"|'[^']*')*\s*src="([^"]*)"`)
```

Comment:
Fixed in https://github.com/cloudquery/cloudquery/pull/16033/commits/d26e88d9d2eb5bb39481d337ee80510f7df90a3e but since Go's regex doesn't have backreferences I can't make sure quote types match each other (I can do it in code but then it would be too late)

Reply:
Removed in https://github.com/cloudquery/cloudquery/pull/16033/commits/015d93250a0e8aa498f7e712e32592b6e997725b in favour of x/net/html